### DISCUSSION TITLE
Can't get the ComponentCommands in samples to work.

### DISCUSSION THREAD

**Discussion by Pieter-JanLavaerts** [spring-projects/spring-shell#588] [General]
Hello,

I would like to use the StringInput, SingleItemSelector and PathInput components in my own project but I can't get them to work.
When I try and use them in the ComponentCommands class of your samples they also don't work so I'm wondering how I'm supposed to use these.
<img width="312" alt="Screenshot 2022-12-09 at 11 19 01" src="https://user-images.githubusercontent.com/42444653/206679832-6285fb8a-a635-49fc-a7cb-e21143a965ed.png">

I think the problem lies with the getTerminal call. This always returns an instance of a DumbTerminal in my case.
When I run this in an actual terminal instead of my IDE I get the same results.

Thank you for reading my issue

**Comment by jvalkeal** [comment]
Hey, moved this to discussions.

Can you give a bit more info? How did you run it. Did you compile it yourself or took jar out from repo.spring.io? What OS is this? Just asking because static screenshot doesn't tell that much.

**Reply by Pieter-JanLavaerts** [reply]
I cloned this repository and opened it up in ItelliJ, then ran the SpringShellSample class on macOS. 
I've also tried making a project with Spring Initializr and adding the following class from the documentation:
```java
@ShellComponent
public class ComponentCommands extends AbstractShellComponent {

    @ShellMethod(key = "component string", value = "String input", group = "Components")
    public String stringInput(boolean mask) {
        StringInput component = new StringInput(getTerminal(), "Enter value", "myvalue");
        component.setResourceLoader(getResourceLoader());
        component.setTemplateExecutor(getTemplateExecutor());
        if (mask) {
            component.setMaskCharater('*');
        }
        StringInputContext context = component.run(StringInputContext.empty());
        return "Got value " + context.getResultValue();
    }
}
```
I ran that on both macOS and Linux.

**Reply by jvalkeal** [reply]
It's a bit unclear if you run it directly from OS console or used it within IntelliJ which has problems in its internal console, see discussion https://github.com/spring-projects/spring-shell/discussions/534#discussioncomment-3662477.

If it fails in a real OS console then I'd like to hear more. If it's just within IntelliJ you need to complain to them.

**Comment by Pieter-JanLavaerts** [comment]
Thanks for responding.
I ran it from an actual terminal and now it works. Sorry for wasting your time.

**Reply by jvalkeal** [reply]
No worries, happy that you got it working.

We probably need to add some notes to docs about IntelliJ as you're not a first one getting into trouble with it's then difficult to figure out why things fail. Getting `DumbTerminal` from JLine was a key as it then throws out all sophisticated shell features as it doesn't know on what environment it's running. 

It's a bit sad that jetbrains don't have that much interest to make it work. Eclipse/STS has similar issues but things partially works. VSCode is probably best IDE to develop cli apps as it can run it within its internal terminal emulation(based on xterm.js).

**Reply by jvalkeal** [reply]
Created #589 for this.
